File: single-language-analysis/dataset-files/java/TestSmell045.java
// Original URL: https://github.com/rdlester/hermes/blob/156c7cdc5479c50e952cea0878728383c1d2eb73/hermesTest/core/BeingTest.java#L93-L105

public class TestSmell045 {
	@Test
	public void test_processUpdate() {
		TestBeing being = new TestBeing();	
		being.setVelocity(makeVector(0.5f,0.0f,0.0f));
		being.updateTime();
		try {
			Thread.sleep(1000);
		} catch (InterruptedException e) { assertTrue(false); }
		being.processUpdate();
		assertEquals(being.getPosition().x, 0.5f, 1e-2);
		assertEquals(being.getPosition().y, 0.0f, 1e-2);
		assertEquals(being.getPosition().z, 0.0f, 1e-2);
	}
}

File: single-language-analysis/dataset-files/python/test_smell_045.py
import pytest

class TestClassname:
    def test_smell_045(self):
        being = TestBeing()    
        being.set_velocity(make_vector(0.5, 0.0, 0.0))
        being.update_time()
        try:
            time.sleep(1)
        except Exception:
            assert False
        being.process_update()
        assert being.get_position().x == pytest.approx(0.5, rel=1e-2)
        assert being.get_position().y == pytest.approx(0.0, rel=1e-2)
        assert being.get_position().z == pytest.approx(0.0, rel=1e-2)